frontends/web/src/routes/account/receive/receive.tsx

```diff
@@ -96,6 +100,18 @@ export const Receive = ({
     }
   }, [addressType, availableScriptTypes, receiveAddresses]);
 
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    if (addressDialog) {
+      e.preventDefault();
+      setActiveIndex(0);
+      setAddressType(addressDialog.addressType);
+      dialogRef.current.deactivate();
+      setTimeout(() => {
+        setAddressDialog(undefined);
+      }, isMobile ? 350 : 0);
```

Comment:
would be nice if the `<Dialog>` component would take care of the timeout.

Reply:
Yes, we won't need this timeout on each component (and firing deactivate) once we do [this](https://github.com/digitalbitbox/bitbox-wallet-app/pull/1842#discussion_r1023451442). :)